src/main.cpp

```diff
@@ -8209,31 +8200,24 @@ void IncrementNeuralNetworkSupermajority(const std::string& NeuralHash, const st
             return;
         }
     }
-    double temp_hashcount = 0;
-    if (mvNeuralNetworkHash.size() > 0)
-    {
-            temp_hashcount = mvNeuralNetworkHash[NeuralHash];
-    }
+
     // 6-13-2015 ONLY Count Each Neural Hash Once per GRC address / CPID (1 VOTE PER RESEARCHER)
-    std::string Security = ReadCache("neuralsecurity",GRCAddress).value;
+    const std::string& Security = ReadCache("neuralsecurity",GRCAddress).value;
     if (Security == NeuralHash)
     {
         //This node has already voted, throw away the vote
         return;
     }
+    
     WriteCache("neuralsecurity",GRCAddress,NeuralHash,GetAdjustedTime());
-    if (temp_hashcount == 0)
-    {
-        mvNeuralNetworkHash.insert(map<std::string,double>::value_type(NeuralHash,0));
-    }
-    double multiplier = 200;
-    if (distance < 40) multiplier = 400;
+
+    // Reference to the value stored within the map to avoid multiple lookups.
+    double& hashcount = mvNeuralNetworkHash[NeuralHash];
+    double multiplier = distance < 40 ? 400 : 200;
     double votes = (1/distance)*multiplier;
-    temp_hashcount += votes;
-    mvNeuralNetworkHash[NeuralHash] = temp_hashcount;
+    hashcount += votes;
 }
 
-
 void IncrementVersionCount(const std::string& Version)
```

Comment:
What about axing this code? Same stats are now available on demand in `getblockstats`.

Reply:
Oh, it's not actually used for anything? Will axe.